lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -731,6 +731,26 @@ LogicalResult WiresOp::verify() {
       return op.emitOpError() << "with name: " << groupName
                               << " is unused in the control execution schedule";
   }
+
+  // Verify that:
+  // - At most one continuous assignment exists for any given value
+  // - A continuously assigned wire has no assignments inside groups.
+  for (auto thisAssignment : getBody()->getOps<AssignOp>()) {
+    if (thisAssignment.guard())
```

Comment:
Maybe a comment here explaining this, e.g.,

```
// Always assume guarded assignments will not be driven simultaneously.
```

I imagine we could statically check simpler cases, such as if some wire is driven when `a` and `!a`, but I think this is fine for now.

Reply:
👍 please add a comment and note that we are liberally (not sure if it's the right term... opposite of conservatively) assuming guards are mutually exclusive. As @cgyurgyik said there could be some interesting static checks. @rachitnigam has also suggested we could emit dynamic checks using sv.assert to ensure the property holds.

Reply:
Yup, the Verilator backend emits such checks and the calyx interpreter dynamically checks this property.